### PR TITLE
Session06 Pull Request

### PR DESCRIPTION
**Pull request by zconn**
Pull request with some other updates as well.

### CODE REVIEW

File: students/ZackConnaughton/session05/mailroom.py [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+
+def enter_donation(response, money):
+    if not response in donors_list:
+        donors_list[response] = []
+    donors_list[response].append(money)
+    return None
+
+
+def save_thank_you_letter(donor, letter_text):
+    f = open('./thank_you_letters/' + donor + '.txt', 'w')
```
Comment (charlesrobison) [inline students/ZackConnaughton/session05/mailroom.py]: This assumes a thank_you_letters directory exists.  What if there isn't one?

File: students/ZackConnaughton/session05/mailroom.py [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+                 'q': quit_menu
+                 }
+    response = ' '
+    while True:
+        print('Select from the following')
+        print('Send single thank you: "t"\n'
+              'Create a Report: "r"\n'
+              'Send letters to all donors: "l"\n'
+              'Quit: "q"')
+        response = input(' => ')[:1].lower()
```
Comment (charlesrobison) [inline students/ZackConnaughton/session05/mailroom.py]: Really like this!

File: students/ZackConnaughton/session05/mailroom.py [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+        if menu_dict.get(response, unknown)() == "quit":
+            break
+
+
+if __name__ == '__main__':
+    main()
```
Comment (charlesrobison) [inline students/ZackConnaughton/session05/mailroom.py]: Works really well!

File: students/ZackConnaughton/session07/html_render.py [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+    tag = "li"
+
+class H(OneLineTag):
+
+    def __init__(self, level, content=None, **kwargs):
+        self.tag = "h" + str(level)
+        super().__init__(content, **kwargs)
+
+class head(Element):
+    tag = "head"
```
Comment (charlesrobison) [inline students/ZackConnaughton/session07/html_render.py]: Looks great - nice job!

File: students/ZackConnaughton/session07/test_html_render.py [hunk 5]
```diff
@@ -0,0 +1,395 @@
...
+
+    # break into lines to check indentation
+    lines = file_contents.split('\n')
+    # making sure the opening and closing tags are right.
+    assert lines[0] == "<html>"
+    # this line should be indented by the amount specified
+    # by the class attribute: "indent"
+    assert lines[1].startswith((" " * Element.indent) + "thi")
+    assert lines[2] == "</html>"
+    assert file_contents.endswith("</html>")
```
Comment (charlesrobison) [inline students/ZackConnaughton/session07/test_html_render.py]: All tests pass

File: students/ZackConnaughton/session08/test_ultimate_circle.py [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+    assert "{:.2f}".format(s.area) == '201.06'
+
+def test_sphere_from_diameter():
+    s = Sphere.from_diameter(8)
+    assert s.radius == 4
+
+def test_sphere_adding():
+    s1 = Sphere(2)
+    s2 = Sphere(3)
+    assert s1 + s2 == Sphere(5)
```
Comment (charlesrobison) [inline students/ZackConnaughton/session08/test_ultimate_circle.py]: Great job - all tests pass.

File: students/ZackConnaughton/session08/ultimate_circle.py [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+    def __repr__(self):
+        return "Sphere({})".format(self.radius)
+
+    @property
+    def volume(self):
+        return (4/3) * math.pi * (self.radius ** 3)
+
+    @property
+    def area(self):
+        return 4 * super().area
```
Comment (charlesrobison) [inline students/ZackConnaughton/session08/ultimate_circle.py]: Great!

**Comment by charlesrobison** [comment]
@zconn  - see if you can resolve the conflict in students/ZackConnaughton/mailroom/mailroom.py so this can be merged.  You may want to simply delete that file.  Try git rm mailroom.py from that directory.

**Comment by zconn** [comment]
> @zconn - looks like there are conflicts that need resolving - reach out if you need help.

@charlesrobison I believe I removed the file from my directory and git. Not sure if it updated the pull though and if I should do another pull request. If that didn't work maybe we can talk Tuesday or I can show you what I have in git and hopefully figure out what is happening.

**Comment by PythonCHB** [comment]
@zconn : what file did you remove? the conflict is in:

students/ZackConnaughton/mailroom/mailroom.py

but if you make a change, and commit it, and then push that to your gitHub repo -- this PR will be automatically updated.

Let's look in class if you can't figure it out.

**Pull request merged**